internal/provider/client.go

```diff
@@ -18,18 +20,72 @@ type ProviderHTTPClient interface {
 	Get(path string) ([]byte, diag.Diagnostics)
 	Update(path string, data io.Reader) ([]byte, diag.Diagnostics)
 	Delete(path string) ([]byte, diag.Diagnostics)
+	setApiEndpoint() diag.Diagnostics
+	getApiEndpoint() string
+}
+
+func GetKeyFromJson[T interface{}](jsonData []byte, keyName string, value *T) error {
```

Comment:
This seems like more of a utility function than a client function, would it makes sense to move it to utils.go?